core/src/main/java/org/apache/calcite/schema/Macro.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.schema;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Schema class for all user defined macros.
+ */
+public class Macro {
```

Comment:
I don't think so as that is used for macros that return tables. It also an interface that has a [function apply](https://github.com/googleinterns/calcite/blob/97399801253a1417ff2676c8740c06180b2e26f1/core/src/main/java/org/apache/calcite/schema/TableMacro.java#L35) that doesn't make sense for this case.

Reply:
I see. So is a TableMacro a subset of a Macro? It seems like these two classes should be linked by inheritance in some way to show their relationship.

Reply:
Since TableMacro is an interface I'm not sure how to connect it up to Macro (which is not an interface). We could make Macro an interface itself but that seems unnecessary since Macro doesn't itself have any interesting properties, it's just there so we can differentiate it between other Function implementations. What do you think?

Reply:
Makes sense. We can leave them as is then.